builder/tests/common-types/src/main/java/io/helidon/common/types/AnnotationBlueprint.java

```diff
@@ -19,38 +19,48 @@
 import java.util.Map;
 import java.util.Optional;
 
+import io.helidon.builder.api.Prototype;
+import io.helidon.config.metadata.ConfiguredOption;
+
 /**
- * Represents an annotation along with its value(s).
+ * An annotation with defined values.
  */
-public interface AnnotationAndValue {
-
+@Prototype.Blueprint
+@Prototype.CustomMethods(AnnotationSupport.class)
+@Prototype.Implement("java.lang.Comparable<Annotation>")
+interface AnnotationBlueprint {
     /**
      * The type name, e.g., {@link java.util.Objects} -> "java.util.Objects".
      *
      * @return the annotation type name
      */
+    @ConfiguredOption(required = true)
     TypeName typeName();
 
+    /**
+     * Get a key-value of all the annotation properties.
+     *
+     * @return key-value pairs of all the properties present
+     */
+    @Prototype.Singular
+    Map<String, String> values();
+
     /**
      * The value property.
      *
      * @return The string value of value property
      */
-    Optional<String> value();
+    default Optional<String> value() {
+        return getValue("value");
+    }
 
     /**
      * Get a value of an annotation property.
      *
      * @param name name of the annotation property
      * @return string value of the property
      */
-    Optional<String> value(String name);
-
-    /**
-     * Get a key-value of all the annotation properties.
-     *
-     * @return key-value pairs of all the properties present
-     */
-    Map<String, String> values();
-
+    default Optional<String> getValue(String name) {
```

Comment:
It cannot be named like that, because that conflicts with a setter for value. As builder has all the getter methods, this was not achievable.
I had a choice - remove `Builder value(String)`, or rename `Optional<String> value(String)`, I have chosen the latter